feat: add method to render as const enums values for go generator.

ibernabeudev opened the pull request:
**Description**

- Add a new func to render enum values as const for Go generator
- Works for integer (using iota) or strings


**Related issue(s)**
See also #647

jonaslagoni commented:
Thanks for taking the time to add this @ibernabeudev!

@smoya would you be able to take a look at the Go output? 🙂

smoya commented on examples/generate-go-enums/__snapshots__/index.spec.ts.snap:
```diff
@@ -0,0 +1,38 @@
+  "// Cities represents an enum of string.
+type Cities string
+
+const (
```
Nitpick: Would it make sense to add a comment here such as: 
```suggestion
// These are all the valid values for Cities enum.
const (
```

In that way, Godoc documentation will add an extra context, useful IMHO:

```bash 
❯ go doc -all main.go


CONSTANTS

const (
        London    Cities = "London"
        Rome             = "Rome"
        Brusseles        = "Brusseles"
)
    These are all the valid values for Cities enum.


TYPES

type Cities string
    Cities represents an enum of string.
```

smoya commented on examples/generate-go-enums/README.md:
```diff
@@ -0,0 +1,17 @@
+# Go Data Models
+
+A basic example of how to Modelina render enums values in Go.
```
```suggestion
A basic example of how Modelina renders enum values in Go.
```

smoya reviewed (commented):
Thanks for your work here @ibernabeudev ! 💯 🚀 

Go output code seems ok to me 👍  Just wrote a nitpick.
JS part review is pending by other folks.

ibernabeudev commented:
Thanks @smoya for your feedback I commit your suggestions. 

I just add a minor change to generate the const names with the type as first word (camel case) to avoid possible collisions when you have different name type enums but with an equal shared value, also I think this is how it was expected to works in the initial issue.

jonaslagoni reviewed (commented):
Do you mind adding the example to https://github.com/asyncapi/modelina/tree/master/examples ? 🙂 

You also have some linting errors 🙂 

Otherwise, it looks good to me 👍

jonaslagoni commented:
/rtm

jonaslagoni reviewed (changes requested):
You have some linting errors 🙂

ibernabeudev commented:
> You have some linting errors 🙂

Fixed, I rewrite the render method because I was having a  **Generic Object Injection Sink (security/detect-object-injection).** annoying linter error.

Also, I added notation **eslint-disable-next-line sonarjs/no-duplicate-string** on line 15 because I saw we used it in another file to avoid have to extract the literal "interface{}" (I think we want to be explicit for these cases).

jonaslagoni commented:
> Fixed, I rewrite the render method because I was having a **Generic Object Injection Sink (security/detect-object-injection).** annoying linter error.

Don't disable them 🙂 You can use `String(xxx)` to fix it.

jonaslagoni reviewed (approved):
LGTM 👍

jonaslagoni reviewed (changes requested):
I see the blackbox tests caught a problem 🙂

ibernabeudev commented:
> I see the blackbox tests caught a problem 🙂
Hi, sorry I was a bit busy these weeks. I fixed the blackbox test error. 

Also I saw a problem with the dockerfile for build the image that runs this test but you can easily solve adding two lines before copy the package-lock.json from hosts to container.

```
....
# Setup library 
WORKDIR /usr/app
COPY package.json .
COPY package-lock.json .
RUN npm install
COPY . .
```

In fact, you could move the last **COPY** command to the next line after **WORKDIR** under the comment **Setup library** and delete the others COPY lines saving two docker layers I think. 

I didn't add this fix to the MR because I don't know if this is correct and I suppose this must be place in another PR.

jonaslagoni commented:
/rtm

jonaslagoni commented:
@all-contributors please add @ibernabeudev for code, test, example, docs
